### ISSUE TITLE
Symbol binding causes errors

### ISSUE THREAD

**Issue by Nycto** [beef331/fungus#5]
First off -- I love this library. Thanks for publishing it.

That said, it looks like naming conflicts with existing symbols throws a wrench in some of the macro magic. I expect this to work, but it currently fails:

```nim
import fungus, macros

adtEnum(Shape):
  None
  Circle: tuple[x, y, r: int]
  Rectangle: tuple[x, y, w, h: int]
  Line: tuple[x1, y1, x2, y2: int]

proc circ*(): string = "foo"

proc myVar*(): string = "foo"

proc myOtherVar*(): string = "foo"

proc rect*(): string = "foo"

proc x1*(): string = "foo"

proc x2*(): string = "foo"

proc `as`*[T](a, b: T): T = discard

proc exec[T]() =
  var a = Shape Circle.init(10, 10, 100)
  a.to(Circle).r = 300
  echo $a.to(Circle)
  a = Shape Line.init(0, 0, 1, 1)

  if (var myVar: Line) from a:
    inc myVar.x1
    echo myVar
  elif (myOtherVar: Circle) from a:
    echo myOtherVar

  match a:
  of Circle as mut circ:
    circ.r = 1000
    echo circ
  of Rectangle as rect:
    echo rect
  of Line as (mut x1, _, x2, _):
    inc x1
    echo a.to(Line)
  else: discard


  if (myVar: Line) from a:
    echo myVar

exec[int]()
```

The error I'm seeing is:

```
/home/nycto/Code/fungus/src/fungus.nim(310, 16) match
/home/nycto/Code/fungus/tests/tsymbols.nim(46, 5) template/generic instantiation of `exec` from here
/home/nycto/Code/fungus/tests/tsymbols.nim(31, 3) template/generic instantiation of `match` from here
/home/nycto/Code/fungus/tests/tsymbols.nim(32, 13) Error: Invaid operation expected 'as'.
```

**Comment by beef331** [comment]
Fixed in 0.1.15 (I take 0 blame, the Nim compiler is giving typed ast for `varargs[untyped]` :smile: )

**Comment by Nycto** [comment]
👏 Thanks for such a quick turn around 👍
